src/async.js

```diff
@@ -117,3 +121,37 @@ export const promiseTimeout = (promise, ms) => {
     });
   });
 };
+
+/**
+ * Returns a Promise which will resolve when `eventName` is triggered on `eventEmitter`.
+ * If `checkFn` is specified, it must return truthy for the Promise to resolve.
+ * Example:
```

Comment:
Note required since you have good tests (remove things that might get out of sync).